From 9eafeefd9f8ea61300e771b836c240b32843a18e Mon Sep 17 00:00:00 2001
From: Larisa Grigore <larisa.grigore@nxp.com>
Date: Fri, 18 Jun 2021 16:52:02 +0300
Subject: [PATCH 16/29] dts: s32gen1: Add recovery pinmuxing for i2c

Add recovery pinmuxing for ERR050686 "I2C: I2C controller
is unable to generate clocks when SDA is low".

Upstream-Status: Pending

Issue: ALB-7193
Signed-off-by: Larisa Grigore <larisa.grigore@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/dts/fsl-s32g274abluebox3.dts | 65 +++++++++++++++++++++++++--
 arch/arm/dts/fsl-s32g274aevb.dts      | 65 +++++++++++++++++++++++++--
 arch/arm/dts/fsl-s32g274ardb.dtsi     | 61 +++++++++++++++++++++++--
 arch/arm/dts/fsl-s32r45.dts           | 31 ++++++++++++-
 4 files changed, 208 insertions(+), 14 deletions(-)

diff --git a/arch/arm/dts/fsl-s32g274abluebox3.dts b/arch/arm/dts/fsl-s32g274abluebox3.dts
index c121c7d19f..28e807073c 100644
--- a/arch/arm/dts/fsl-s32g274abluebox3.dts
+++ b/arch/arm/dts/fsl-s32g274abluebox3.dts
@@ -5,6 +5,7 @@
  */
 
 /dts-v1/;
+#include <dt-bindings/gpio/gpio.h>
 #include "fsl-s32g274a.dtsi"
 
 / {
@@ -44,8 +45,11 @@
 
 &i2c0 {
 	clock-frequency=<100000>;
+	pinctrl-names = "default", "gpio";
 	pinctrl-0 = <&pinctrl0_i2c0>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&pinctrl0_i2c0_gpio>;
+	scl-gpios = <&gpio0 17 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio0 16 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 
 	eeprom@50 {
@@ -67,22 +71,31 @@
 
 &i2c1 {
 	clock-frequency=<100000>;
+	pinctrl-names = "default", "gpio";
 	pinctrl-0 = <&pinctrl0_i2c1 &pinctrl1_i2c1>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&pinctrl0_i2c1_gpio &pinctrl1_i2c1_gpio>;
+	scl-gpios = <&gpio0 19 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio0 20 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
 &i2c2 {
 	clock-frequency=<100000>;
+	pinctrl-names = "default", "gpio";
 	pinctrl-0 = <&pinctrl0_i2c2 &pinctrl1_i2c2>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&pinctrl0_i2c2_gpio &pinctrl1_i2c2_gpio>;
+	scl-gpios = <&gpio0 21 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio0 22 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
 &i2c4 {
 	clock-frequency=<100000>;
+	pinctrl-names = "default", "gpio";
 	pinctrl-0 = <&pinctrl0_i2c4 &pinctrl1_i2c4>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&pinctrl0_i2c4_gpio &pinctrl1_i2c4_gpio>;
+	scl-gpios = <&gpio0 34 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio0 33 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 
 	vr5510 {
@@ -127,24 +140,50 @@
 				    >;
 		};
 
+		pinctrl0_i2c0_gpio: pinctrl0_i2c0_gpio {
+			fsl,pins = <PB00_MSCR_S32G PB00_I2C0_SDA_GPIO
+				    PB01_MSCR_S32G PB01_I2C0_SCL_GPIO
+				    I2C0_SDA_IMCR IMCR_DISABLED
+				    I2C0_SCL_IMCR IMCR_DISABLED
+				    >;
+		};
+
 		pinctrl0_i2c1: pinctrl0_i2c1 {
 			fsl,pins = <PB03_MSCR_S32G PB03_I2C1_SCL_CFG
 				    PB04_MSCR_S32G PB04_I2C1_SDA_CFG
 				    >;
 		};
 
+		pinctrl0_i2c1_gpio: pinctrl0_i2c1_gpio {
+			fsl,pins = <PB03_MSCR_S32G PB03_I2C1_SCL_GPIO
+				    PB04_MSCR_S32G PB04_I2C1_SDA_GPIO
+				    >;
+		};
+
 		pinctrl0_i2c2: pinctrl0_i2c2 {
 			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_CFG
 				    PB06_MSCR_S32G PB06_I2C2_SDA_CFG
 				    >;
 		};
 
+		pinctrl0_i2c2_gpio: pinctrl0_i2c2_gpio {
+			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_GPIO
+				    PB06_MSCR_S32G PB06_I2C2_SDA_GPIO
+				    >;
+		};
+
 		pinctrl0_i2c4: pinctrl0_i2c4 {
 			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_CFG
 				    PC02_MSCR_S32G PC02_I2C4_SCL_CFG
 				    >;
 		};
 
+		pinctrl0_i2c4_gpio: pinctrl0_i2c4_gpio {
+			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_GPIO
+				    PC02_MSCR_S32G PC02_I2C4_SCL_GPIO
+				    >;
+		};
+
 		pinctrl0_sd0: pinctrl0_sd0 {
 			fsl,pins = <PC14_MSCR_S32G PC14_SD0_CLK_CFG
 				    PC15_MSCR_S32G PC15_SD0_CMD_CFG
@@ -192,16 +231,34 @@
 				    >;
 		};
 
+		pinctrl1_i2c1_gpio: pinctrl1_i2c1_gpio {
+			fsl,pins = <I2C1_SCL_IMCR IMCR_DISABLED
+				    I2C1_SDA_IMCR IMCR_DISABLED
+				    >;
+		};
+
 		pinctrl1_i2c2: pinctrl1_i2c2 {
 			fsl,pins = <I2C2_SCL_IMCR PB05_I2C2_SCL_IN
 				    I2C2_SDA_IMCR PB06_I2C2_SDA_IN
 				    >;
 		};
 
+		pinctrl1_i2c2_gpio: pinctrl1_i2c2_gpio {
+			fsl,pins = <I2C2_SCL_IMCR IMCR_DISABLED
+				    I2C2_SDA_IMCR IMCR_DISABLED
+				    >;
+		};
+
 		pinctrl1_i2c4: pinctrl1_i2c4 {
 			fsl,pins = <I2C4_SDA_IMCR PC01_I2C4_SDA_IN
 				    I2C4_SCL_IMCR PC02_I2C4_SCL_IN
 				    >;
 		};
+
+		pinctrl1_i2c4_gpio: pinctrl1_i2c4_gpio {
+			fsl,pins = <I2C4_SDA_IMCR IMCR_DISABLED
+				    I2C4_SCL_IMCR IMCR_DISABLED
+				    >;
+		};
 	};
 };
diff --git a/arch/arm/dts/fsl-s32g274aevb.dts b/arch/arm/dts/fsl-s32g274aevb.dts
index 5d0908d665..10201eb20b 100644
--- a/arch/arm/dts/fsl-s32g274aevb.dts
+++ b/arch/arm/dts/fsl-s32g274aevb.dts
@@ -4,6 +4,7 @@
  */
 
 /dts-v1/;
+#include <dt-bindings/gpio/gpio.h>
 #include "fsl-s32g274a.dtsi"
 
 / {
@@ -84,32 +85,44 @@
 /* EEPROM */
 &i2c0 {
 	clock-frequency=<100000>;
+	pinctrl-names = "default", "gpio";
 	pinctrl-0 = <&pinctrl0_i2c0>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&pinctrl0_i2c0_gpio>;
+	scl-gpios = <&gpio0 17 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio0 16 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
 /* Platform board PCI X16 EXPRESS - I2C_SCL_S0, I2C_SDA_S0  */
 &i2c1 {
 	clock-frequency=<100000>;
+	pinctrl-names = "default", "gpio";
 	pinctrl-0 = <&pinctrl0_i2c1 &pinctrl1_i2c1>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&pinctrl0_i2c1_gpio &pinctrl1_i2c1_gpio>;
+	scl-gpios = <&gpio0 19 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio0 20 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
 /* Platform board GPIO_J3-17 (SDA), GPIO_J3-19 (SCL0)  */
 &i2c2 {
 	clock-frequency=<100000>;
+	pinctrl-names = "default", "gpio";
 	pinctrl-0 = <&pinctrl0_i2c2 &pinctrl1_i2c2>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&pinctrl0_i2c2_gpio &pinctrl1_i2c2_gpio>;
+	scl-gpios = <&gpio0 21 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio0 22 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
 /* PMIC */
 &i2c4 {
 	status = "okay";
+	pinctrl-names = "default", "gpio";
 	pinctrl-0 = <&pinctrl0_i2c4 &pinctrl1_i2c4>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&pinctrl0_i2c4_gpio &pinctrl1_i2c4_gpio>;
+	scl-gpios = <&gpio0 34 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio0 33 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	clock-frequency=<100000>;
 
 	vr5510 {
@@ -136,24 +149,50 @@
 				    >;
 		};
 
+		pinctrl0_i2c0_gpio: pinctrl0_i2c0_gpio {
+			fsl,pins = <PB00_MSCR_S32G PB00_I2C0_SDA_GPIO
+				    PB01_MSCR_S32G PB01_I2C0_SCL_GPIO
+				    I2C0_SDA_IMCR IMCR_DISABLED
+				    I2C0_SCL_IMCR IMCR_DISABLED
+				    >;
+		};
+
 		pinctrl0_i2c1: pinctrl0_i2c1 {
 			fsl,pins = <PB03_MSCR_S32G PB03_I2C1_SCL_CFG
 				    PB04_MSCR_S32G PB04_I2C1_SDA_CFG
 				    >;
 		};
 
+		pinctrl0_i2c1_gpio: pinctrl0_i2c1_gpio {
+			fsl,pins = <PB03_MSCR_S32G PB03_I2C1_SCL_GPIO
+				    PB04_MSCR_S32G PB04_I2C1_SDA_GPIO
+				    >;
+		};
+
 		pinctrl0_i2c2: pinctrl0_i2c2 {
 			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_CFG
 				    PB06_MSCR_S32G PB06_I2C2_SDA_CFG
 				    >;
 		};
 
+		pinctrl0_i2c2_gpio: pinctrl0_i2c2_gpio {
+			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_GPIO
+				    PB06_MSCR_S32G PB06_I2C2_SDA_GPIO
+				    >;
+		};
+
 		pinctrl0_i2c4: pinctrl0_i2c4 {
 			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_CFG
 				    PC02_MSCR_S32G PC02_I2C4_SCL_CFG
 				    >;
 		};
 
+		pinctrl0_i2c4_gpio: pinctrl0_i2c4_gpio {
+			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_GPIO
+				    PC02_MSCR_S32G PC02_I2C4_SCL_GPIO
+				    >;
+		};
+
 		pinctrl0_qspi: pinctrl0_qspi {
 			fsl,pins = <PF05_MSCR_S32G PF05_QSPI_DATA_A0_CFG
 				    PF06_MSCR_S32G PF06_QSPI_DATA_A1_CFG
@@ -243,18 +282,36 @@
 				    >;
 		};
 
+		pinctrl1_i2c1_gpio: pinctrl1_i2c1_gpio {
+			fsl,pins = <I2C1_SCL_IMCR IMCR_DISABLED
+				    I2C1_SDA_IMCR IMCR_DISABLED
+				    >;
+		};
+
 		pinctrl1_i2c2: pinctrl1_i2c2 {
 			fsl,pins = <I2C2_SCL_IMCR PB05_I2C2_SCL_IN
 				    I2C2_SDA_IMCR PB06_I2C2_SDA_IN
 				    >;
 		};
 
+		pinctrl1_i2c2_gpio: pinctrl1_i2c2_gpio {
+			fsl,pins = <I2C2_SCL_IMCR IMCR_DISABLED
+				    I2C2_SDA_IMCR IMCR_DISABLED
+				    >;
+		};
+
 		pinctrl1_i2c4: pinctrl1_i2c4 {
 			fsl,pins = <I2C4_SDA_IMCR PC01_I2C4_SDA_IN
 				    I2C4_SCL_IMCR PC02_I2C4_SCL_IN
 				    >;
 		};
 
+		pinctrl1_i2c4_gpio: pinctrl1_i2c4_gpio {
+			fsl,pins = <I2C4_SDA_IMCR IMCR_DISABLED
+				    I2C4_SCL_IMCR IMCR_DISABLED
+				    >;
+		};
+
 		pinctrl1_dspi1: pinctrl1_dspi1 {
 			fsl,pins = <DSPI1_SIN_IMCR PF15_SPI1_SIN_IN
 				    >;
diff --git a/arch/arm/dts/fsl-s32g274ardb.dtsi b/arch/arm/dts/fsl-s32g274ardb.dtsi
index 5b14d58662..828e3f6fa9 100644
--- a/arch/arm/dts/fsl-s32g274ardb.dtsi
+++ b/arch/arm/dts/fsl-s32g274ardb.dtsi
@@ -4,6 +4,7 @@
  */
 
 /dts-v1/;
+#include <dt-bindings/gpio/gpio.h>
 #include "fsl-s32g274a.dtsi"
 
 &usdhc0 {
@@ -34,32 +35,44 @@
 /* IO Expander  */
 &i2c0 {
 	clock-frequency=<100000>;
+	pinctrl-names = "default", "gpio";
 	pinctrl-0 = <&pinctrl0_i2c0>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&pinctrl0_i2c0_gpio>;
+	scl-gpios = <&gpio0 32 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio0 31 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
 /* PCIe X1 Connector  */
 &i2c1 {
 	clock-frequency=<100000>;
+	pinctrl-names = "default", "gpio";
 	pinctrl-0 = <&pinctrl1_i2c1>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&pinctrl1_i2c1_gpio>;
+	scl-gpios = <&gpio1 163 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio1 165 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
 /* FLEXRAY_LIN  */
 &i2c2 {
 	clock-frequency=<100000>;
+	pinctrl-names = "default", "gpio";
 	pinctrl-0 = <&pinctrl0_i2c2 &pinctrl1_i2c2>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&pinctrl0_i2c2_gpio &pinctrl1_i2c2_gpio>;
+	scl-gpios = <&gpio0 21 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio0 22 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
 /* PMIC */
 &i2c4 {
 	clock-frequency=<100000>;
+	pinctrl-names = "default", "gpio";
 	pinctrl-0 = <&pinctrl0_i2c4 &pinctrl1_i2c4>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&pinctrl0_i2c4_gpio &pinctrl1_i2c4_gpio>;
+	scl-gpios = <&gpio0 34 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio0 33 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 
 	vr5510 {
@@ -85,18 +98,38 @@
 				    >;
 		};
 
+		pinctrl0_i2c0_gpio: pinctrl0_i2c0_gpio {
+			fsl,pins = <PB15_MSCR_S32G PB15_I2C0_SDA_GPIO
+				    PC00_MSCR_S32G PC00_I2C0_SCL_GPIO
+				    I2C0_SDA_IMCR IMCR_DISABLED
+				    I2C0_SCL_IMCR IMCR_DISABLED
+				    >;
+		};
+
 		pinctrl0_i2c2: pinctrl0_i2c2 {
 			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_CFG
 				    PB06_MSCR_S32G PB06_I2C2_SDA_CFG
 				    >;
 		};
 
+		pinctrl0_i2c2_gpio: pinctrl0_i2c2_gpio {
+			fsl,pins = <PB05_MSCR_S32G PB05_I2C2_SCL_GPIO
+				    PB06_MSCR_S32G PB06_I2C2_SDA_GPIO
+				    >;
+		};
+
 		pinctrl0_i2c4: pinctrl0_i2c4 {
 			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_CFG
 				    PC02_MSCR_S32G PC02_I2C4_SCL_CFG
 				    >;
 		};
 
+		pinctrl0_i2c4_gpio: pinctrl0_i2c4_gpio {
+			fsl,pins = <PC01_MSCR_S32G PC01_I2C4_SDA_GPIO
+				    PC02_MSCR_S32G PC02_I2C4_SCL_GPIO
+				    >;
+		};
+
 		pinctrl0_qspi: pinctrl0_qspi {
 			fsl,pins = <PF05_MSCR_S32G PF05_QSPI_DATA_A0_CFG
 				    PF06_MSCR_S32G PF06_QSPI_DATA_A1_CFG
@@ -180,18 +213,38 @@
 				    >;
 		};
 
+		pinctrl1_i2c1_gpio: pinctrl1_i2c1_gpio {
+			fsl,pins = <PK03_MSCR_S32G PK03_I2C1_SCL_GPIO
+				    PK05_MSCR_S32G PK05_I2C1_SDA_GPIO
+				    I2C1_SCL_IMCR IMCR_DISABLED
+				    I2C1_SDA_IMCR IMCR_DISABLED
+				    >;
+		};
+
 		pinctrl1_i2c2: pinctrl1_i2c2 {
 			fsl,pins = <I2C2_SCL_IMCR PB05_I2C2_SCL_IN
 				    I2C2_SDA_IMCR PB06_I2C2_SDA_IN
 				    >;
 		};
 
+		pinctrl1_i2c2_gpio: pinctrl1_i2c2_gpio {
+			fsl,pins = <I2C2_SCL_IMCR IMCR_DISABLED
+				    I2C2_SDA_IMCR IMCR_DISABLED
+				    >;
+		};
+
 		pinctrl1_i2c4: pinctrl1_i2c4 {
 			fsl,pins = <I2C4_SDA_IMCR PC01_I2C4_SDA_IN
 				    I2C4_SCL_IMCR PC02_I2C4_SCL_IN
 				    >;
 		};
 
+		pinctrl1_i2c4_gpio: pinctrl1_i2c4_gpio {
+			fsl,pins = <I2C4_SDA_IMCR IMCR_DISABLED
+				    I2C4_SCL_IMCR IMCR_DISABLED
+				    >;
+		};
+
 		pinctrl1_dspi1: pinctrl1_dspi1 {
 			fsl,pins = <DSPI1_SIN_IMCR PF15_SPI1_SIN_IN
 				    >;
diff --git a/arch/arm/dts/fsl-s32r45.dts b/arch/arm/dts/fsl-s32r45.dts
index 0658f318c6..d03f4300c3 100644
--- a/arch/arm/dts/fsl-s32r45.dts
+++ b/arch/arm/dts/fsl-s32r45.dts
@@ -9,6 +9,7 @@
  */
 #include <dt-bindings/clock/s32r45-clock.h>
 #include <dt-bindings/clock/s32r45-scmi-clock.h>
+#include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/pinctrl/s32r45-pinctrl.h>
 
 /dts-v1/;
@@ -225,16 +226,22 @@
 /* J36 - PMIC */
 &i2c0 {
 	clock-frequency=<100000>;
+	pinctrl-names = "default", "gpio";
 	pinctrl-0 = <&pinctrl0_i2c0>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&pinctrl0_i2c0_gpio>;
+	scl-gpios = <&gpio0 17 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio0 16 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
 /* J37 */
 &i2c1 {
 	clock-frequency=<100000>;
+	pinctrl-names = "default", "gpio";
 	pinctrl-0 = <&pinctrl0_i2c1 &pinctrl1_i2c1>;
-	pinctrl-names = "default";
+	pinctrl-1 = <&pinctrl0_i2c1_gpio &pinctrl1_i2c1_gpio>;
+	scl-gpios = <&gpio0 14 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
+	sda-gpios = <&gpio0 15 (GPIO_ACTIVE_HIGH | GPIO_OPEN_DRAIN)>;
 	status = "okay";
 };
 
@@ -305,12 +312,26 @@
 				    >;
 		};
 
+		pinctrl0_i2c0_gpio: pinctrl0_i2c0_gpio {
+			fsl,pins = <PB00_MSCR_S32R45 PB00_I2C0_SDA_GPIO
+				    PB01_MSCR_S32R45 PB01_I2C0_SCL_GPIO
+				    I2C0_SDA_IMCR IMCR_DISABLED
+				    I2C0_SCL_IMCR IMCR_DISABLED
+				    >;
+		};
+
 		pinctrl0_i2c1: pinctrl0_i2c1 {
 			fsl,pins = <PA14_MSCR_S32R45 PA14_I2C1_SCL_CFG
 				    PA15_MSCR_S32R45 PA15_I2C1_SDA_CFG
 				    >;
 		};
 
+		pinctrl0_i2c1_gpio: pinctrl0_i2c1_gpio {
+			fsl,pins = <PA14_MSCR_S32R45 PA14_I2C1_SCL_GPIO
+				    PA15_MSCR_S32R45 PA15_I2C1_SDA_GPIO
+				    >;
+		};
+
 		pinctrl0_qspi: pinctrl0_qspi {
 			fsl,pins = <PF05_MSCR_S32R45 PF05_QSPI_DATA_A0_CFG
 				    PF06_MSCR_S32R45 PF06_QSPI_DATA_A1_CFG
@@ -397,6 +418,12 @@
 				    I2C1_SDA_IMCR PA15_I2C1_SDA_IN
 				    >;
 		};
+
+		pinctrl1_i2c1_gpio: pinctrl1_i2c1_gpio {
+			fsl,pins = <I2C1_SCL_IMCR IMCR_DISABLED
+				    I2C1_SDA_IMCR IMCR_DISABLED
+				    >;
+		};
 	};
 };
 
-- 
2.17.1

